### ISSUE TITLE
Support `@hideconstructor` JSDoc tag to hide constructors from classes.

### ISSUE THREAD

**Issue by typhonrt** [TypeStrong/typedoc#2577]
For classes that have only static methods presently there is no clean way to prevent TypeDoc from adding a default constructor to docs generated. This feature proposal suggests that the JSDoc tag [@hideconstructor](https://jsdoc.app/tags-hideconstructor) is useful. There is not an equivalent tag in TSDoc as far as I'm aware. 

```js
/**
 * This class should not be constructed as it only contains static methods.
 * 
 * @hideconstructor
 */
export class StaticClass {
   static foo() { ... }
}
```

This tag can also be applied on constructors as well per the JSDoc description. 

---

TSC won't output default constructors even if they have comments / JSDoc attached to them. 

The only non-clean workaround that I've found is to provide a constructor with a parameter and use the `@hidden` tag to remove it from documentation. This is very non-clean because in generated type declarations the dummy constructor is exposed.

--- 

I can take a look at a PR for this if you are too busy as this feature improves the API docs for libraries I release.

**Comment by Gerrit0** [comment]
That's no class, it's a namespace in disguise!

Because of that, it should really have a private constructor, which TS will emit in declaration files.

```ts
export class StaticClass {
   private constructor() {}
   static foo() { ... }
}
```

If `--excludePrivate` is set, that'll result in the constructor automatically being hidden, and removes the need for the clarifying comment as the type system enforces the desired behavior. 

---

A [search](https://github.com/search?q=%22%40hideconstructor%22+%28language%3AJavaScript+OR+language%3ATypeScript+%29&type=code) on GitHub shows this is used 9.1k times, which is more than I expected, but still much less than the 123k for [`@ignore`](https://github.com/search?q=%22%40ignore%22+%28language%3AJavaScript+OR+language%3ATypeScript+%29&type=code) / 56.9k for [`@hidden`](https://github.com/search?q=%22%40hidden%22+%28language%3AJavaScript+OR+language%3ATypeScript+%29&type=code)

It feels like usage of this tag would really indicate something else is wrong to me... that said, that's the state of JS in general, so I'm begrudgingly starting to think that TypeDoc should support this...

---

(wait, what, `--excludePrivate` defaults to false?! going to change that with 0.26...)

**Comment by Gerrit0** [comment]
That behavior definitely smells like a bug to me. TypeScript is incorrectly representing the type of your class in the declaration file. Possible I'm missing something, we'll see [what the TS team says](https://github.com/microsoft/TypeScript/issues/58653)

It's an easy patch to add it, so I went ahead and did it, but will definitely be discouraging its use in the documentation.

**Comment by typhonrt** [comment]
Thanks for following up with the issue in the TS repo. Indeed this definitely seems like a TS issue and if resolved would server the purpose far better than `@hideconstructor`.  I also just noticed that if one does the following and _doesn't_ add `@param` for the bogus parameter `_invalid` that it will generate desired types:

```js
export class StaticClass {
   /**
    * @private
    */
   constructor(_invalid) {
      throw new Error('This is a static class and should not be constructed.');
   }
}
```

```js
declare class StaticClass {
  /**
   * @private
   */
  private constructor();
}

**Comment by Gerrit0** [comment]
Included in TypeDoc 0.26, which is releasing 2024/06/21
